lib/jackhammer/topic.rb

```diff
@@ -21,10 +21,16 @@ def queues
 
       @queues = @queue_config.map do |name, options|
         handler = MessageReceiver.new(options.delete('handler'))
-        routing = options.delete 'routing_key'
-        queue = Jackhammer.channel.queue name, options
-        Queue.new topic: @topic, queue: queue, handler: handler, routing: routing
+        routing = fetch_and_delete_key(options, 'routing_key')
+        queue = Jackhammer.channel.queue(name, options)
```

Comment:
How about not mutating the options at all: 
```
handler = MessageReceiver.new(options.fetch('handler'))
routing = options.fetch('routing_key')
```
and then it is clear from the look at single line, which options are skipped here etc 
```
queue = Jackhammer.channel.queue(name, options.without('routing_key', 'handler'))
```

Reply:
I'm fine either way. I'll suggest a third alternative 👯‍♂ which would be to have the exception raised in line so that the diff is only +/- 1 since we're not using the method anywhere else (yet). Cosmetics aside, it's a good idea to halt process booting if a queue doesn't have a routing_key defined 👍 
```patch
- routing = options.delete 'routing_key'
+ routing = options.delete 'routing_key' || fail(InvalidConfigError, "routing_key not found in #{options.inspect}")
```

Reply:
Good call on not mutating the options. Unfortunately  `without` is an ActiveSupport method, which is not  a jackhammer dependency. I think we should introduce `QueueConfig` class (or actually `SubscriptionConfig`) abstracting the config and taking care of validations and filtering native options which are passed to `Jackhammer.channel.queue`.

Reply:
Fact ! I forgot no `AS` here. We could go with `Hash#slice` then for a quick win. Though `SubscriptionConfig` value object is even better.

Reply:
I just noticed it is merged already. cosmetic thing anyway.